pkg/apis/componentconfig/types.go

```diff
@@ -62,3 +62,213 @@ const (
 	ProxyModeUserspace ProxyMode = "userspace"
 	ProxyModeIPTables  ProxyMode = "iptables"
 )
+
+type KubeletConfiguration struct {
```

Comment:
In general for config based options omit empty reduces readability - most
config files today document every field they have as part of the config
file on disk (apache, haproxy, nginx, redis, etc).  I would think we would
do the same.

On Tue, Jan 12, 2016 at 8:17 PM, Phillip Wittrock <notifications@github.com>
wrote:

> In pkg/apis/componentconfig/types.go
> <https://github.com/kubernetes/kubernetes/pull/19507#discussion_r49540516>
> :
>
> > @@ -62,3 +62,213 @@ const (
> >  	ProxyModeUserspace ProxyMode = "userspace"
> >  	ProxyModeIPTables  ProxyMode = "iptables"
> >  )
> > +
> > +type KubeletConfiguration struct {
>
> per-Brian's comment, a few of these have empty defaults. Is it possible to
> add 'omitempty' to the json tag for those?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/19507/files#r49540516>.
>